plugins/hikaricp/src/main/java/com/navercorp/pinpoint/plugin/hikaricp/HikariCpPlugin.java

```diff
@@ -72,26 +62,88 @@ private void addBasicDataSourceTransformer() {
+
+        // 1.1.1 ~ 2.4.1
+        transformTemplate.transform("com.zaxxer.hikari.proxy.ConnectionProxy", new ConnectionTransformCallback());
+    }
+
+    private void addHikariPoolTransformer() {
+        // 1.3.7 ~ 2.6.x (without 2.3.x)
+        transformTemplate.transform("com.zaxxer.hikari.pool.HikariPool", new HikariPoolTransformCallback());
+
+        // 2.3.x
+        transformTemplate.transform("com.zaxxer.hikari.pool.BaseHikariPool", new HikariPoolTransformCallback());
+    }
+
     @Override
     public void setTransformTemplate(TransformTemplate transformTemplate) {
         this.transformTemplate = transformTemplate;
     }
 
-    static class ConnectionTransformCallback implements TransformCallback {
+    private static class ConnectionTransformCallback implements TransformCallback {
 
         @Override
         public byte[] doInTransform(Instrumentor instrumentor, ClassLoader classLoader, String className, Class<?> classBeingRedefined, ProtectionDomain protectionDomain, byte[] classfileBuffer) throws InstrumentException {
             InstrumentClass target = instrumentor.getInstrumentClass(classLoader, className, classfileBuffer);
-            target.addInterceptor("com.navercorp.pinpoint.plugin.hikaricp.interceptor.DataSourceCloseInterceptor");
+            target.addInterceptor(HikariCpConstants.INTERCEPTOR_CLOSE_CONNECTION);
             return target.toBytecode();
         }
 
     }
+
+    private static class HikariPoolTransformCallback implements TransformCallback {
+
+        @Override
+        public byte[] doInTransform(Instrumentor instrumentor, ClassLoader classLoader, String className, Class<?> classBeingRedefined, ProtectionDomain protectionDomain, byte[] classfileBuffer) throws InstrumentException {
+            InstrumentClass target = instrumentor.getInstrumentClass(classLoader, className, classfileBuffer);
+            if (isAvailableDataSourceMonitor(target)) {
+                // ~ 2.4.0
+                InstrumentMethod constructor = target.getConstructor("com.zaxxer.hikari.HikariConfig", "java.lang.String", "java.lang.String");
+                if (constructor != null) {
+                    addDataSourceMonitorInterceptor(target, constructor);
+                    return target.toBytecode();
+                }
+
+                // 2.4.1 ~
+                constructor = target.getConstructor("com.zaxxer.hikari.HikariConfig");
+                if (constructor != null) {
+                    addDataSourceMonitorInterceptor(target, constructor);
+                    return target.toBytecode();
+                }
+            }
+            return target.toBytecode();
+        }
+
+        private boolean isAvailableDataSourceMonitor(InstrumentClass target) {
+            InstrumentMethod getActiveConnections = target.getDeclaredMethod("getActiveConnections");
+            if (getActiveConnections == null || !int.class.getName().equals(getActiveConnections.getReturnType())) {
```

Comment:
It is better to consider the case where the result value of getActiveConnections.getReturnType () is null.
JavassistMethod.java
~~~
    @Override
    public String getReturnType() {
        if (behavior instanceof CtMethod) {
            try {
                return ((CtMethod) behavior).getReturnType().getName();
            } catch (NotFoundException e) {
                return null;
            }
        }

        return null;
    }
~~~